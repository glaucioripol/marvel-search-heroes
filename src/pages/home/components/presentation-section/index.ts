export { PresentationSection } from "./presentation-section";
