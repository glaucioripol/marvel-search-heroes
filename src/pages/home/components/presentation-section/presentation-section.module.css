.presentation_section {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.presentation_section .presentation_section__title {
  color: var(--grey-2);
  font-size: var(--font-xl);
  text-transform: uppercase;
}

.presentation_section .presentation_section__description {
  color: var(--grey-3);
  font-size: var(--font-x-sm);
}

.presentation_section
  .presentation_section__title
  + .presentation_section__description {
  margin-block-start: 16px;
}
