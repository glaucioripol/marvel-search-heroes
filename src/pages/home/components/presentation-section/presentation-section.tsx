import styles from "./presentation-section.module.css";

export function PresentationSection() {
  return (
    <section className={styles.presentation_section}>
      <h1
        className={styles.presentation_section__title}
        data-testid="presentation-title">
        explore o universo
      </h1>

      <p
        className={styles.presentation_section__description}
        data-testid="presentation-description">
        Mergulhe no domínio deslumbrante de todos os personagens clássicos que
        você ama - e aqueles que você descobrirá em breve!
      </p>
    </section>
  );
}
