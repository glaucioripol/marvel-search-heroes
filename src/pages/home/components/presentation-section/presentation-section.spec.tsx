import {
  describe,
  it,
  expect,
  beforeEach,
  defaultRender,
  screen,
} from "@/__tests__";

import { PresentationSection } from "./presentation-section";

describe("PresentationSection component", () => {
  let component: ReturnType<typeof defaultRender>;

  beforeEach(() => {
    component = defaultRender(<PresentationSection />);
  });

  it("should render title correctly", () => {
    const titleElement = screen.getByTestId("presentation-title");
    expect(titleElement).toBeInTheDocument();
    expect(titleElement.textContent).toBe("explore o universo");
  });

  it("should render description correctly", () => {
    const descriptionElement = screen.getByTestId("presentation-description");
    expect(descriptionElement).toBeInTheDocument();
    expect(descriptionElement.textContent).deep.equal(
      "Mergulhe no domínio deslumbrante de todos os personagens clássicos que você ama - e aqueles que você descobrirá em breve!",
    );
  });

  it("should match snapshot", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
