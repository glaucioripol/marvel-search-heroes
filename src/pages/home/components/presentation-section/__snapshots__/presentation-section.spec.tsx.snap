// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PresentationSection component > should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="_presentation_section_e38c7f"
  >
    <h1
      class="_presentation_section__title_e38c7f"
      data-testid="presentation-title"
    >
      explore o universo
    </h1>
    <p
      class="_presentation_section__description_e38c7f"
      data-testid="presentation-description"
    >
      Mergulhe no domínio deslumbrante de todos os personagens clássicos que você ama - e aqueles que você descobrirá em breve!
    </p>
  </section>
</DocumentFragment>
`;
